pdns/recursor_cache.cc

```diff
@@ -288,16 +308,16 @@ int32_t MemRecursorCache::get(time_t now, const DNSName &qname, const QType& qt,
         *state = *cachedState;
       }
 
-      return ret > 0 ? static_cast<int32_t>(ret-now) : ret;
+      return ret > 0 ? fakeTTD(nullptr, qname, qtype, ret, now, origTTL, refresh) : ret;
```

Comment:
Does that means that `ADDR` queries (for the A and AAAA of NS) will not be added to the refresh list? I'm OK with that if that's the case, but does it still makes sense to return a fake TTD? Does that allow other entries to be refreshed?

Reply:
The `nullptr` is there to avoid having to keep track which cache entry was used. You are right that in this case using the fakeTTD isn;t of much use. This also means that the first arg of can be a ref since it will always be there